src/domain/entities/DataSet.ts

```diff
@@ -0,0 +1,100 @@
+import { Id, Ref } from "$/domain/entities/Ref";
+import { Maybe } from "$/utils/ts-utils";
+import { BasicDataSet, BasicDataSetAttrs, D2Translation } from "$/domain/entities/BasicDataSet";
+
+export interface DataSetAttrs extends BasicDataSetAttrs {
+    name: string;
+    displayFormName: string;
+    formName: string;
+    sections: Section[];
+    dataSetElements: DataSetElement[];
```

Comment:
I think you only need to specify "sections", all the other come from the parent class.